pydicom/filewriter.py

```diff
@@ -383,49 +389,54 @@ def write_data_element(fp, data_element, encoding=default_encoding):
             fp.write(VR)
         if VR in extra_length_VRs:
             fp.write_US(0)  # reserved 2 bytes
-    if VR not in writers:
-        raise NotImplementedError(
-            "write_data_element: unknown Value Representation '{0}'".format(
-                VR))
 
     # write into a buffer to avoid seeking back which can be expansive
     buffer = DicomBytesIO()
     buffer.is_little_endian = fp.is_little_endian
     buffer.is_implicit_VR = fp.is_implicit_VR
 
-    encoding = convert_encodings(encoding)
-    writer_function, writer_param = writers[VR]
-    if VR in text_VRs:
-        writer_function(buffer, data_element, encoding=encoding[1])
-    elif VR in ('PN', 'SQ'):
-        writer_function(buffer, data_element, encoding=encoding)
-    else:
-        # Many numeric types use the same writer but with numeric format
-        # parameter
-        if writer_param is not None:
-            writer_function(buffer, data_element, writer_param)
+    if data_element.is_raw:
+        if (fp.is_implicit_VR == data_element.is_implicit_VR and
+                fp.is_little_endian == data_element.is_little_endian):
+            buffer.write(data_element.value)
+            is_undefined_length = data_element.length == 0xFFFFFFFF
+        else:
+            data_element = DataElement_from_raw(data_element, encoding)
+    if not data_element.is_raw:
```

Comment:
Couldn't this just be an `else` block?

Reply:
No, because it could have been converted above.